Tweak lang-mustache factory

dakrone opened the pull request:
This commits makes a few very minor tweaks to our Mustache scripting capabilities.

1. Switches from `DefaultMustacheFactory` to `SafeMustacheFactory`
In the event that the security manager were disabled (or removed, as is threatened in subsequent JDK releases), Mustache's "partial template" (`{{>partial}}`) support is a security risk because it would allow reading from an arbitrary URL or file on disk. Switching to the "Safe" version and passing in an empty set into the parent constructor disallows using partial Mustache templates. This also switches from `Function<String, String>` to the `TemplateFunction` value for the built-in functions.

3. Minor internal optimization
This removes useless grouping for one of the built-in mustache functions, and removes the call to `CollectionUtils.ensureNoSelfReferences` used in `CustomReflectionObjectHandler`. This check during stringification should not be necessary because Mustache templates without "partial" support cannot be self-referencing.

dakrone committed:
Tweak lang-mustache factory

This commits makes a few very minor tweaks to our Mustache scripting capabilities.

1. Switches from `DefaultMustacheFactory` to `SafeMustacheFactory`
In the event that the security manager were disabled (or removed, as is threatened in subsequent JDK releases), Mustache's "partial template" (`{{>partial}}`) support is a security risk because it would allow reading from an arbitrary URL or file on disk. Switching to the "Safe" version and passing in an empty set into the parent constructor disallows using partial Mustache templates. This also switches from `Function<String, String>` to the `TemplateFunction` value for the built-in functions.

2. Minor internal optimizations
This removes useless grouping for one of the built-in mustache functions, and removes the call to `CollectionUtils.ensureNoSelfReferences` used in `CustomReflectionObjectHandler`. This check during stringification should not be necessary because Mustache templates without "partial" support cannot be self-referencing.

elasticsearchmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

grcevski reviewed (approved):
LGTM! This looks good to me, but I don't know all the history of our mustache support.

dakrone committed:
Fix Mustache throwing transform exceptions instead Watcher

Previously we threw the exception from Mustache, rather than checking the transform script itself again.

dakrone committed:
Merge remote-tracking branch 'origin/main' into mustache-experiments

dakrone commented:
@elasticmachine run elasticsearch-ci/part-3
